mmv1/products/vertexai/DeploymentResourcePool.yaml

```diff
@@ -0,0 +1,175 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'DeploymentResourcePool'
+base_url: 'projects/{{project}}/locations/{{region}}/deploymentResourcePools'
+create_url: 'projects/{{project}}/locations/{{region}}/deploymentResourcePools'
+self_link: 'projects/{{project}}/locations/{{region}}/deploymentResourcePools/{{deployment_resource_pool_id}}'
+# No method is not defined for DeploymentResourcePools
+immutable: true
+create_verb: :POST
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.deploymentResourcePools
+async: !ruby/object:Api::OpAsync
+  actions:
+    - create
+    - delete
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+description: |-
+  'DeploymentResourcePool can be shared by multiple deployed models,
+  whose underlying specification consists of dedicated resources.'
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'vertex_ai_deployment_resource_pool'
+    primary_resource_id: 'deployment_resource_pool'
+    # Skip the import test because deployment_resource_pool_id can't be specified in an auto-generated test
+    skip_import_test: true
```

Comment:
[QUESTION] Should I add a manually written test for importing a resource by specifying `ImportId:`?

Reply:
Ah, the import format we're creating is wrong- it assumes that `/{{name}}` is appended to the `base_url`. If we add an `identity` entry similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/ce2d84b886cbd808d650624170695b564128d861/mmv1/products/storage/BucketAccessControl.yaml#L42-L43, this should generate import formats that are correct & compatible with the resource id (which is used as the import key)

For this resource that value'll be `deploymentResourcePoolId`

Reply:
Thank you for your comment! I updated the code to use `name` as identifier along with the custom encoder in [99bb911](https://github.com/GoogleCloudPlatform/magic-modules/pull/9518/commits/99bb911307b70377267cd15ebd7d230edabfdfee). So `skip_import_test` and `identity` became unnecessary in the end.